neps/nep-0508.md

```diff
@@ -66,17 +66,51 @@ Currently, NEAR protocol has four shards. With more partners onboarding, we star
 
 TBD. e.g. configuration changes we have to introduce
 
+A new protocol version will be introduced specifying the new shard layout. 
+
 ### Required state changes
 
 TBD. e.g. additional/updated data a node has to maintain
 
+* For the duration of the resharding the node will need to maintain a snapshot of the flat state and related columns.
+* For the duration of the epoch before the new shard layout takes effect, the node will need to maintain the state and flat state of shards in the old and new layout at the same time. 
+
 ### Resharding flow
 
 TBD. how resharding happens at the high level
 
+* The new shard layout will be agreed on offline by the protocol team and hardcoded in the neard reference implementation.
+* In epoch T the protocol version upgrade date will pass and nodes will vote to switch to the new protocol version. The new protocol version will contain the new shard layout. 
+* In epoch T, in the last block of the epoch, the EpochConfig for epoch T+2 will be set. The EpochConfig for epoch T+2 will have the new shard layout.
+* In epoch T + 1, all nodes will perform the state split. The child shards will be kept up to date with the blockchain up until the epoch end.
+* In epoch T + 2, the chain will switch to the new shard layout. 
+
 ## Reference Implementation
```

Comment:
may want to add some code snippet (OR link to the code) in case reviewers want to know more about actual implementation details.